lms/djangoapps/certificates/tests/test_support_views.py

```diff
@@ -326,6 +333,34 @@ def test_regenerate_user_has_no_certificate(self):
         num_certs = GeneratedCertificate.eligible_certificates.filter(user=self.student).count()
         self.assertEqual(num_certs, 1)
 
+    def test_regenerate_user_with_invalidated_cert(self):
+        """ Verify if the certificate is marked as invalid, if so, regenerate
+        the certificate and de-activate the invalidated entry. """
+
+        # mark certificate as invalid
+        self._invalidate_certificate(self.cert)
+        self.assertTrue(self._invalidate_cert_exists())
+        # after invalidation certificate become un-available.
+        self.assertTrue(
+            GeneratedCertificate.objects.filter(  # pylint: disable=no-member
+                user=self.student, status=CertificateStatuses.unavailable
+            ).exists()
```

Comment:
You are doing this twice make a method and pass user and status as arguments.Something like this 

```
def is_generated_certificate_exist(user, status):
    GeneratedCertificate.objects.filter(
        user=user, status=status
    ).exists()
```